### PR TITLE
*: add autocommit judgment and test

### PR DESCRIPTION
**Pull request by zimulala**

### CODE REVIEW

File: sessionctx/variable/session.go [hunk 1]
```diff
@@ -94,18 +94,15 @@ func (s *SessionVars) AddFoundRows(rows uint64) {
...
-}
-
-// SetStatusInTrans sets the status flags about ServerStatusInTrans.
-func (s *SessionVars) SetStatusInTrans(isInTrans bool) {
-	if isInTrans {
-		s.Status |= mysql.ServerStatusInTrans
+// SetStatus sets the session server status variable.
+// If flag is ture sets the input status in session status,
+// otherwise removes the input status.
+func (s *SessionVars) SetStatus(flag bool, status uint16) {
```
Comment (coocood) [inline sessionctx/variable/session.go]: I think rename as the following would be better
`SetStatusFlag(flag uint16, on bool)`

Comment (zimulala) [inline sessionctx/variable/session.go]: got it.

**Comment by coocood** [comment]
LGTM

**Comment by ngaut** [comment]
LGTM

**Pull request merged**